File: problems/easy/Arrays/1929. Concatenation of Array/concatenation_of_array.py
from typing import List
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        # O(n) time | O(2 * n) space
        ans = [0] * 2 * len(nums)
        for i in range(len(nums)):
            ans[i], ans[i + len(nums)] = nums[i], nums[i]
        return ans

File: problems/easy/Arrays/1929. Concatenation of Array/concatenation_of_array.cpp
#include <vector>

using std::vector;

class Solution {
public:
    // O(n) time | O(2 * n) space
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> ans(nums.size() * 2, 0);
        for(int i = 0; i < nums.size(); i++) {
            ans[i] = nums[i];
            ans[i + nums.size()] = nums[i];
        }
        return ans;
    }
};